src/meta/src/hummock/manager/mod.rs

```diff
@@ -1160,85 +1270,126 @@ where
                 level_deltas.push(LevelDelta {
                     delta_type: Some(DeltaType::GroupDestroy(GroupDestroy {})),
                 });
+                new_version_delta.gc_sst_ids.append(&mut gc_sst_ids);
                 new_hummock_version.levels.remove(&group_id);
             }
         }
+        let mut new_groups = vec![];
         // these `group_id`s must be unique
         for (
             group_id,
             CompactionGroup {
                 compaction_config, ..
             },
-        ) in compaction_groups
+        ) in compaction_groups.iter()
         {
-            if !new_hummock_version.levels.contains_key(&group_id) {
+            if !new_hummock_version.levels.contains_key(group_id) {
                 new_hummock_version
                     .levels
                     .try_insert(
-                        group_id,
-                        <Levels as HummockLevelsExt>::build_initial_levels(&compaction_config),
+                        *group_id,
+                        <Levels as HummockLevelsExt>::build_initial_levels(compaction_config),
                     )
                     .unwrap();
+                new_groups.push(*group_id);
+                need_independent_commit = true;
                 let level_deltas = &mut new_version_delta
                     .level_deltas
-                    .entry(group_id)
+                    .entry(*group_id)
                     .or_default()
                     .level_deltas;
                 level_deltas.push(LevelDelta {
                     delta_type: Some(DeltaType::GroupConstruct(GroupConstruct {
-                        group_config: Some(compaction_config),
+                        group_config: Some(compaction_config.clone()),
                     })),
                 });
             }
         }
+
+        if !need_independent_commit {
```

Comment:
> Looks like `need_independent_commit == false` only happens when there is no compaction group change. Instead of introducing this flag, can we calculate the diff between `old_version_groups` and `compaction_groups` at the beginning of `sync_group`, and early return if the diff is empty to make the code more readable?

fixed.